### DISCUSSION TITLE
Curious why `beforeFirst` in `ResultSet` is not supported?

### DISCUSSION THREAD

**Discussion by xxks-kkk** [duckdb/duckdb#6489] [Q&A]
So, essentially, in DuckDB, for every result set I get, I can only iterate through it once (If I need to use the same ResultSet somehwere else, I have to re-execute the same statement, which is very inefficient). It seems like the only workaround is to use `CachedRowSet` (https://github.com/duckdb/duckdb/pull/4070/files#diff-7e79d886d5e5e16e577d9e329b460f6d163cf71f8daba8c8acb46ffbd22d5408)? Is `CachedRowSet` fully functioned? In my application, I manage `ResultSet` object directly and it would be very expensive to switch to `CachedRowSet` in order to work with DuckDB.

Thank you

**Comment by hannes** [comment]
Should be straightforward to support this , happy to review a PR
